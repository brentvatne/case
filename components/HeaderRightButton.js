import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

import colors from '../constants/Colors'
import layout from '../constants/Layout'
import type from '../constants/Type'

const styles = StyleSheet.create({
  button: {
    paddingRight: layout.padding,
    paddingHorizontal: layout.padding,
  },
  text: {
    color: colors.gray.text,
    fontSize: type.sizes.normal,
    fontWeight: 'bold',
  },
})

export default class HeaderRightButton extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isSaving: false,
    }
    this.onPress = this.onPress.bind(this)
  }

  onPress() {
    if (this.state.isSaving) return false
    this.setState({ isSaving: true })
    return this.props.onPress()
  }

  render() {
    const { text } = this.props

    return (
      <TouchableOpacity onPress={this.onPress} style={styles.button}>
        <Text style={styles.text}>{text}</Text>
      </TouchableOpacity>
    )
  }
}

HeaderRightButton.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func,
}

HeaderRightButton.defaultProps = {
  onPress: () => null,
}
