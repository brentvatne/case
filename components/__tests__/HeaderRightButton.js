import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import HeaderRightButton from '../HeaderRightButton'

test('renders correctly', () => {
  const tree = shallow(
    <HeaderRightButton text="Done" />,
  )
  expect(toJSON(tree)).toMatchSnapshot()
})
