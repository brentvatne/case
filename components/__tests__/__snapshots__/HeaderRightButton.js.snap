// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
  style={
    Object {
      "paddingHorizontal": 10,
      "paddingRight": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#585858",
        "fontSize": 12,
        "fontWeight": "bold",
      }
    }
  >
    Done
  </Text>
</TouchableOpacity>
`;
